pkg/openstack/common.go

```diff
@@ -7,12 +7,32 @@ import (
 
 	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
 	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
 	apimeta "k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// CheckDeleteDisabled - Delete the object which in turn will clean the sub resources
+func CheckDeleteDisabled(ctx context.Context, helper *helper.Helper, obj client.Object) (ctrl.Result, error) {
+	key := client.ObjectKeyFromObject(obj)
+	if err := helper.GetClient().Get(ctx, key, obj); err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+	// Delete the object
+	if obj.GetDeletionTimestamp().IsZero() {
+		if err := helper.GetClient().Delete(ctx, obj); err != nil {
+			return ctrl.Result{}, err
```

Comment:
`err` can be NotFound here as well (I think) but we should ignore that error.

Reply:
I can add that, but we just checked that it exists before calling delete.

Reply:
as L20 and L28 is two separate API requests I believe it can happen that the first sees the object but second does not see it any more. It is probably a scenario with low probability but having a check for NotFound here is also cheap. I would add that check.

Reply:
Yeah, different API requests but they are sequential and Get is not asynchronous like delete. I can add it though.